### PR TITLE
Implemented palindrome partitioning algorithm

### PR DESCRIPTION
**Pull request by KomoderHell**
palindrome partitioning algorithm: you are given a string and you need to find out the minimum number of partitions required, such the resulting strings are all palindromes.

example:
string = "nitik"

answer = 2

explanation: n | i t i | n

### CODE REVIEW

**Comment by sujaykaushik008** [comment]
@varunu28, please review this PR

File: .idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-997) [inline .idea/jarRepositories.xml:1]: Is this new file necessary???

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package dynamic_programming
+
+/**
+ * Palindrome Partitioning Algorithm
+ *
+ * You are given a string as input, and task is to find the minimum number of partitions to be made,
+ * in the string sot that the resulting strings are all palindrome
+ *
+ * Time Complexity = O(n^2)
+ *
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:10]: Add about author 
@author [name](github url)

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package dynamic_programming
+
+/**
+ * Palindrome Partitioning Algorithm
+ *
+ * You are given a string as input, and task is to find the minimum number of partitions to be made,
+ * in the string sot that the resulting strings are all palindrome
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:7]: Can you elaborate it with some examples?

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * Time Complexity = O(n^2)
+ *
+ * */
+
+
+/**
+ * @param String is the string to be checked
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:15]: Add 1-2 line description for this function

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        if (string[l + i] != string[j - l]) {
+            return false
+        }
+    }
+    return true
+}
+
+
+/**
+ * @param String is the string to be checked
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:31]: Add 1-2 line description for this function

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ * @param Int is the starting index of the string in consideration
+ * @param Int is the ending index of the string in consideration
+ * @return minimum number of partitions required
+ **/
+fun palindromePartition(string: String, i: Int, j: Int): Int {
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:36]: give meaningful names to variables

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * @param String the string on which algorithm is to be operated
+ */
+fun initialize(string: String): Int {
+
+    dp = Array(string.length) { Array(string.length) { -1 } }
+    return palindromePartition(string, 0, string.length - 1)
+}
+
```
Comment (rishabh-997) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:72]: Remove extra lines

**Review by rishabh-997 — CHANGES_REQUESTED** [review]
Address the following changes.

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 2 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:10]: according to contribution guidelines, we are supposed to do that, our commit is sufficient to mark our contribution

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 3 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:7]: the suggested changes have been made, thank you for your insight

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 4 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:16]: I believe the name isPalindrome is self-explaining enough, writing a description would be over-explaining the obvious

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 5 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:32]: As mentioned in the above suggestion, I am avoiding over-explaining the function

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 6 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:37]: I have used generic names for indices(i and j), and given a proper description of them, I have done this in order to keep the variable names.

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt (hunk 7 above)
Comment (sujaykaushik008) [reply src/main/kotlin/dynamic_programming/PalindromePartitioning.kt]: The suggested changes have been made, thank you for your insight

**Comment by sujaykaushik008** [comment]
@varunu28  please, review this PR

**Comment by KomoderHell** [comment]
Can someone review this PR?

File: src/main/kotlin/dynamic_programming/PalindromePartitioning.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
+package dynamic_programming
```
Comment (vbrazo) [inline src/main/kotlin/dynamic_programming/PalindromePartitioning.kt:1]: Is the palindrome algorithm using the dynamic programming technique?

**Comment by vbrazo** [comment]
This seems to be more an Array/String case.

**Review by vbrazo — APPROVED** [review]

**Pull request merged**